game/scripts/vscripts/items/drums_of_endurance.lua

```diff
@@ -0,0 +1,259 @@
+LinkLuaModifier( "modifier_intrinsic_multiplexer", "modifiers/modifier_intrinsic_multiplexer.lua", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_drums_of_endurance_oaa", "items/drums_of_endurance.lua", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_drums_of_endurance_oaa_swiftness_aura", "items/drums_of_endurance.lua", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_drums_of_endurance_oaa_swiftness_aura_effect", "items/drums_of_endurance.lua", LUA_MODIFIER_MOTION_NONE )
+LinkLuaModifier( "modifier_item_drums_of_endurance_oaa_active", "items/drums_of_endurance.lua", LUA_MODIFIER_MOTION_NONE )
+
+item_drums_of_endurance_oaa = class(ItemBaseClass)
+
+function item_drums_of_endurance_oaa:GetIntrinsicModifierName()
+  return "modifier_intrinsic_multiplexer"
+end
+
+function item_drums_of_endurance_oaa:GetIntrinsicModifierNames()
+  return {
+    "modifier_item_drums_of_endurance_oaa",
+    "modifier_item_drums_of_endurance_oaa_swiftness_aura",
+    "modifier_item_drums_of_endurance_oaa_swiftness_aura_effect"
+  }
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+--On Casting/Activating Item
+
+function item_drums_of_endurance_oaa:OnSpellStart()
+  --Initializing needed variables
+  local ability = self
+  local caster = ability:GetCaster()
+  local casterTeam = caster:GetTeamNumber()
+
+  local units = FindUnitsInRadius(
+    casterTeam,
+    caster:GetAbsOrigin(),
+    nil,
+    ability:GetSpecialValueFor("radius"),
+    DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+    bit.bor(DOTA_UNIT_TARGET_BASIC, DOTA_UNIT_TARGET_HERO),
+    DOTA_UNIT_TARGET_FLAG_INVULNERABLE,
+    FIND_ANY_ORDER,
+    false
+  )
+
+  local function EnduranceActive(unit)
+    unit:AddNewModifier(self:GetCaster(), self, "modifier_item_drums_of_endurance_oaa_active", {duration = self:GetSpecialValueFor("duration")})
```

Comment:
Sorry to kind of waste a lot of this work, but pretty much only this part is necessary. I think the vanilla modifiers can all be reused instead of needing new ones. You should just need to use `modifier_item_ancient_janggo` as the intrinsic modifier and apply `modifier_item_ancient_janggo_active` here for the active.

Reply:
-While true, it could help in the future when separating the custom effects and actives from the default ones. (In case changes are to be made, like adding/removing effects and actives).
-I mean if there is a downside to using custom modifiers over vanilla ones please let me know and I will make changes respectively.

Reply:
There isn't necessarily a huge downside, especially for something relatively simple as this. It just guarantees correctness. On that note, your tooltips for the modifiers do not match the vanilla one.

```
"DOTA_Tooltip_modifier_item_ancient_janggo_aura_effect"	"Swiftness Aura"
"DOTA_Tooltip_modifier_item_ancient_janggo_aura_effect_Description"	"Movement speed increased by %dMODIFIER_PROPERTY_MOVESPEED_BONUS_CONSTANT%."
"DOTA_Tooltip_modifier_item_ancient_janggo_active"	"Endurance"
"DOTA_Tooltip_modifier_item_ancient_janggo_active_Description"	"Movement speed increased by %dMODIFIER_PROPERTY_MOVESPEED_BONUS_PERCENTAGE%%%.\nAttack speed increased by %dMODIFIER_PROPERTY_ATTACKSPEED_BONUS_CONSTANT%."
```

Reply:
Changed now to use keys